blueprints/component/index.js

```diff
@@ -21,13 +25,45 @@ module.exports = {
     }
   ],
 
+  filesPath: function() {
+    let filesDirectory = 'files';
+
+    if (isModuleUnificationProject(this.project)) {
+      filesDirectory = 'module-unification-files';
+    }
+
+    return path.join(this.path, filesDirectory);
+  },
+
   fileMapTokens: function() {
+    let isModuleUnification = isModuleUnificationProject(this.project);
+
     return {
+      __root__(options) {
+        if (isModuleUnification) {
```

Comment:
I think this check will break `--dummy` and `--in-repo-addon` cli flags. So if we have a module unification layout then
```
  ember g component some-component --dummy 
```
will generate component in the `src/` instead of `dummy/` dir.

Reply:
Hm... I've just realized that dummy apps has to support module unification layout as well 🤔

Reply:
Thanks, yes. I have a `Add support for addons` todo item above and I've [asked a question](https://embercommunity.slack.com/archives/C5JN29NTC/p1515352731000031) on #st-module-unification looking for an example MU addon so that I can add support for it too